static/js/filter.js

```diff
@@ -382,13 +382,48 @@ Filter.prototype = {
     },
 
     can_mark_messages_read: function () {
-        return _.every(this._operators, function (elem) {
-            return (_.contains(['stream', 'topic', 'pm-with'], elem.operator)
-                || elem.operator === 'is' && elem.operand === 'private'
-                || elem.operator === 'in' && elem.operand === 'all'
-                || elem.operator === 'in' && elem.operand === 'home')
-                && !elem.negated;
```

Comment:
I think you lost the rule about negation -- we want `-pm-with:foo` to return false.  I think that just means adding a check at the start looking for any negated terms, and rejecting if there are any.

(And I guess we should make our tests handle that).

Reply:
Yeah, the story with negated here is the same as the other commit that you recently merged--it's handled by virtue of Filter.term_type making negated terms be "not-pm-with".

But, yeah, we should add a test here.